rfc/rfc000-package-server.md

```diff
@@ -1,19 +1,87 @@
+# Package Server
+
 ---
+
 RFC: 000
-Author: Peter Bourgon <peter@bourgon.org>
+
+Author: Matt Heath <matt@mattheath.com>
+
 Status: Draft
----
 
-# package server
+---
 
 ## Motivation
 
-http://peter.bourgon.org/go-kit/#package-server
+[http://peter.bourgon.org/go-kit/#package-server](http://peter.bourgon.org/go-kit/#package-server)
+
+Package server is probably the biggest and most important component of the toolkit. Ideally, we should be able to write our services as implementations of normal, nominal Go interfaces, and delegate integration with the environment to the server package. The package should encode and enforce conventions for server-side concerns, like health checks, system-wide request tracing, connection management, backpressure and throttling, and so on. For each of those topics, it should provide interfaces for different, pluggable strategies. It should integrate with service discovery, and work equally well over multiple transports. Considerable prior art exists in the form of Finagle, Karyon (Netflixs application service library), and likely many more.
 
 ## Scope
 
-To be defined.
+### Endpoints
+
+*   An endpoint is defined as a handler interface which receives a request and returns a response or an error to the client.
+*   A server SHALL have one or more endpoints.
+*   An endpoint SHALL accept a request context and propagate this - allowing access to request-specific values such as the identity of the end user, authorization tokens, and the request's deadline.
```

Comment:
>  I suppose it doesn't hurt very much to include it.

I'm afraid it might, actually. I'm afraid it conflates the responsibilities of the caller vs. the callee.

> It looks to me like the request context is basically just headers when on the wire and those are marshalled in to the net/context type (or some equivalent) when the request is received in a service. Is this the correct way to view it?

More or less. For me, the canonical example of what goes into the context is the [request trace ID](https://github.com/peterbourgon/gokit/blob/master/rfc/rfc006-request-tracing.md). Stuff that isn't part of the business-level contract of the service, but is nevertheless important to thread through a request tree.